Sources/Sentry/SentryProfiler.mm

```diff
@@ -87,7 +84,10 @@ - (instancetype)init
 #    endif // SENTRY_HAS_UIKIT
 
     [self start];
-    [self scheduleTimeoutTimer];
+
+    if (mode == SentryProfilerModeLegacy) {
+        [self scheduleTimeoutTimer];
+    }
```

Comment:
`m`: What's stopping us from moving the `scheduleTimeoutTimer` logic to the `SentryLegacyProfiler`. Then, this class wouldn't even know that there are different profiling modes, and we could ditch the `SentryProfilerMode` enum.

Reply:
Yes I did consider that. Nothing is stopping it, it should be fine enough to move the timer to a static object in the two control classes.